src/providers/postgres/qgspostgresschemaselectiondialog.h

```diff
@@ -0,0 +1,43 @@
+/***************************************************************************
+    qgspostgresschemaselectiondialog.h
+    ---------------------
+    begin                : January 2025
+    copyright            : (C) 2025 by Jan Caha
+    email                : jan.caha at outlook dot com
+ ***************************************************************************
+ *                                                                         *
+ *   This program is free software; you can redistribute it and/or modify  *
+ *   it under the terms of the GNU General Public License as published by  *
+ *   the Free Software Foundation; either version 2 of the License, or     *
+ *   (at your option) any later version.                                   *
+ *                                                                         *
+ ***************************************************************************/
+#ifndef QGSPOSTGRESSCHEMASELECTIONDIALOG_H
+#define QGSPOSTGRESSCHEMASELECTIONDIALOG_H
+
+#include <QDialog>
+#include <QComboBox>
+#include <QDialogButtonBox>
+
+#include "qgspostgresprojectstorage.h"
+#include "qgsdatasourceuri.h"
+
+class QgsPostgresSchemaSelectionDialog : public QDialog
```

Comment:
There's a large amount of postgres specific changes here which I think could potentially be generalized to work with other project storage providers, but lets defer that till later...

For now, could you generalize at least this dialog to a generic QgsDatabaseSchemaSelectionDialog class? It should rely on the existing QgsDatabaseSchemaComboBox class

Reply:
Done :+1: